nats.go

```diff
@@ -1844,7 +1848,11 @@ func (nc *Conn) doReconnect(err error) {
 		if sleepTime <= 0 {
 			runtime.Gosched()
 		} else {
-			time.Sleep(time.Duration(sleepTime))
+			rt.Reset(time.Duration(rand.Int63n(sleepTime)))
```

Comment:
Reconnect Wait time should still hold same semantics. Do not attempt reconnect for some period of time. What I think we need to add is a Jitter option which has a sane default, I might say 2 secs or so that is added onto reconnectWait.

Reply:
So random between [0,2sec] added to the base, ok got it. Will update.

Reply:
Argg.. quite a bit of tests seem to expect no more than reconnectWait, so we may get some flappers pop up, will have to fix those along the way... unless you think that added jitter should be an option?

Reply:
I don't think it should be an option, but should be low as default. For tests we can set to 0 which should do old behavior.

Reply:
expect that some of those tests are in /test package so I cannot play that trick without exposing an option or function to set that value to zero..

Reply:
Jitter was always meant to be a public option, just wanted a sane default. For the case we were dealing with today they would probably set it to 5-10 secs.. I think that is alot for default cases where you have live backup servers and want quick reconnect. Maybe 1 second is plenty in that case, so [0,1] is default.

Reply:
Ok so we do make it an Option in that it is settable. Still, it means that all tests in /test that relied on no more than reconnectWait will have to be altered, but that's ok. Just that I may not find them all at once.

Reply:
Yes they will need to set Jitter to 0.. ;)